cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+        hidden_field = HiddenFileInput(locator='.//input[contains(@accept,".csv")]')
+
+        table = Table(
+            './/div[contains(@class, "container-fluid")]/table',
+            column_widgets={
+                "Select": Checkbox(locator=".//input"),
+                1: Text('.//span/button[contains(@class,"btn btn-link")]'),
+            },
+        )
+        filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu_vms_step")
+        search_box = TextInput(
+            locator='.//div[contains(@class, "modal-content")]'
+            '//div[contains(@class,"input-group")]/input'
+        )
+        clear_filters = Text(".//a[text()='Clear All Filters']")
+        error_text = Text(
+            './/h3[contains(@class,"blank-slate-pf-main-action") and '
+            'contains(text(), "Error:")]'
+        )
+        error_icon = Text(locator='.//span[contains(@class, "pficon-error-circle-o")]')
+        popover_text = Text(locator='.//div[contains(@class, "popover-content")]')
+
+        @property
+        def is_displayed(self):
+            return self.table.is_displayed and (
+                len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0
+            )
+
+        def csv_import(self, vm_list):
+            temp_file = tempfile.NamedTemporaryFile(suffix=".csv")
+            with open(temp_file.name, "w") as file:
+                headers = ["Name", "Provider"]
+                writer = csv.DictWriter(file, fieldnames=headers)
+                writer.writeheader()
+                for vm in vm_list:
+                    writer.writerow({"Name": vm.name, "Provider": vm.provider.name})
+            self.hidden_field.fill(temp_file.name)
+
+        def fill(self, values):
+            csv_import = values.get('csv_import')
+            vm_list = values.get('vm_list')
+            if csv_import:
+                self.csv_import(vm_list)
+            for vm in vm_list:
+                self.filter("VM Name", vm.name)
+                for row in self.table.rows():
+                    if vm.name in row.vm_name.read():
+                        row[0].fill(True)
+                self.clear_filters.click()
+            self.parent.next_btn.click()
+
+        def after_fill(self, was_change):
+            self.browser.send_keys(Keys.ENTER, self.search_box)
+
+        def filter(self, type, name):
+            try:
+                self.filter_by_dropdown.item_select(type)
+            except NoSuchElementException:
+                logger.info("{} not present in filter dropdown!".format(type))
+            self.search_box.fill(name)
+            self.after_fill(was_change=True)
+
+        def select_by_name(self, vm_name):
+            self.filter("VM Name", vm_name)
+            vms_selected = []
+            for row in self.table.rows():
+                if vm_name in row.read()["VM Name"]:
+                    row.select.fill(True)
+                    vms_selected.append(row.read()["VM Name"])
```

Comment:
Reading the row again is expensive, I think you should assign the row read to a variable, and then reference here and above in the `if` check.